packages/frontend-2/lib/common/helpers/graphql.ts

```diff
@@ -499,52 +529,182 @@ export const getDateCursorFromReference = (params: {
+ * Build skipLoggingErrors function that skips logging errors if there's only one error and it's related to a specific field
+ */
+export const skipLoggingErrorsIfOneFieldError =
+  (fieldName: string | string[]) =>
+  (err: ErrorResponse): boolean => {
+    const fieldNames = isArray(fieldName) ? fieldName : [fieldName]
+    return (
+      err.graphQLErrors?.length === 1 &&
+      err.graphQLErrors.some((e) => intersection(e.path || [], fieldNames).length > 0)
+    )
+  }
+
+type NonUndefined<T> = T extends undefined ? never : T
+
+/**
+ * Update field at specific path in object, only if it exists. Useful for cache modification
+ * when fields should only be updated if they exist.
+ */
+export const updatePathIfExists = <Value, Path extends Paths<Value> & string>(
+  val: Value,
+  path: Path,
+  updater: (val: NonUndefined<Get<Value, Path>>) => NonUndefined<Get<Value, Path>>
+) => {
+  if (!val) return val
+
+  if (has(val, path)) {
+    const pathVal = get(val, path) as NonUndefined<Get<Value, Path>>
+    const newVal = updater(pathVal)
+    set(val, path, newVal)
+  }
+
+  return val
+}
+
+/**
+ * Get value from specific path in object, only if it exists.
+ */
+export const getFromPathIfExists = <Value, Path extends Paths<Value> & string>(
+  val: MaybeNullOrUndefined<Value>,
+  path: Path
+): Optional<Get<Value, Path>> => {
+  if (!val) return undefined
+  if (!has(val, path)) return undefined
+  return get(val, path) as Get<Value, Path>
+}
+
+type ModifyObjectFieldValue<
+  Type extends keyof AllObjectTypes,
+  Field extends keyof AllObjectTypes[Type]
+> = ModifyFnCacheData<AllObjectTypes[Type][Field]>
+
+/**
+ * Simplified & improved version of modifyObjectFields, just targetting a single field for a cache modification
  * @see modifyObjectFields
  */
 export const modifyObjectField = <
-  FieldData = unknown,
-  Variables extends Optional<Record<string, unknown>> = undefined
+  Type extends keyof AllObjectTypes,
+  Field extends keyof AllObjectTypes[Type]
 >(
   cache: ApolloCache<unknown>,
-  id: string,
-  fieldName: string,
+  key: ApolloCacheObjectKey<Type>,
+  fieldName: Field,
   updater: (params: {
     fieldName: string
-    variables: Variables
-    value: ModifyFnCacheData<FieldData>
-    details: Parameters<Modifier<ModifyFnCacheData<FieldData>>>[1] & {
+    variables: Field extends keyof AllObjectFieldArgTypes[Type]
+      ? AllObjectFieldArgTypes[Type][Field]
+      : never
+    /**
+     * Value found in the cache. Read-only and should not be mutated directly. Use the
+     * createUpdatedValue() helper to build a new value with updated fields.
+     */
+    value: ReadonlyDeep<ModifyObjectFieldValue<Type, Field>>
+    helpers: {
+      /**
+       * Build new value with the values at specific paths updated with the provided updater functions,
+       * ONLY if the paths exist in the cache.
+       *
+       * This function operates on a deeply cloned value that is safe to mutate
+       */
+      createUpdatedValue: (
```

Comment:
Simplifies this:
```ts
     modifyObjectField(
                cache,
                workspaceCacheId,
                'projects',
                ({ value, helpers: { ref } }) => {
                  const newItems = isUndefined(value?.items)
                    ? undefined
                    : [ref('Project', newProject.id), ...value.items]

                  const newTotalCount = isUndefined(value?.totalCount)
                    ? undefined
                    : (value.totalCount || 0) + 1

                  return {
                    ...value,
                    ...(isUndefined(newItems) ? {} : { items: newItems }),
                    ...(isUndefined(newTotalCount) ? {} : { totalCount: newTotalCount })
                  }
                }
              )
```

Into this:
```ts
modifyObjectField(
                cache,
                workspaceCacheId,
                'projects',
                ({ helpers: { ref, createUpdatedValue } }) => {
                  return createUpdatedValue(({ update }) => {
                    update('items', (items) => [
                      ref('Project', newProject.id),
                      ...items
                    ])
                    update('totalCount', (count) => count + 1)
                  })
                }
              )
```

update() calls are properly typed as well, and will only allow you to specify property paths that actually exist in `value`, and the property value in the callback is gonna be properly typed too
![image](https://github.com/user-attachments/assets/b1af187a-89ed-493d-a9c8-b3220964dcc8)
![image](https://github.com/user-attachments/assets/f89fd3cc-ee57-4d5e-82f4-53f7da7e31a6)